### PR TITLE
Base content

### PR DESCRIPTION
**Pull request by tswicegood**
Creating a base content model that can be used for cross-model querying.

### CODE REVIEW

File: armstrong/core/arm_content/models.py [hunk 1]
```diff
@@ -1,4 +1,19 @@
...
+from django.db.models.query import QuerySet
+from model_utils.managers import InheritanceManager
+from taggit.managers import TaggableManager
+
+from . import mixins
+
+
+class ContentBase(mixins.AuthorsMixin, mixins.PublicationMixin, models.Model):
+    title = models.CharField(max_length=255)
+    summary = models.TextField()
```
Comment (tswicegood) [inline armstrong/core/arm_content/models.py:29]: Not sure how I feel about `title`/`summary` always being present.

File: armstrong/core/arm_content/models.py [hunk 2]
```diff
@@ -1,4 +1,19 @@
...
+
+    tags = TaggableManager()
```
Comment (tswicegood) [inline armstrong/core/arm_content/models.py:36]: This is tentative.  From a DRY standpoint, it'd be nice to have it here.

Comment (nosamanuel) [inline armstrong/core/arm_content/models.py:36]: Maybe it would be worth having a setting for the default publication mixin class?

Comment (tswicegood) [inline armstrong/core/arm_content/models.py:36]: Possibly...  At first glance I'm ok with having the defaults be the defaults and if you want more you can change them yourself.  For all intents and purposes, `PublicationMixin` is what you should be using and it should have `sites`.  If you deviate from that, you probably want more control anyhow.

**Comment by tswicegood** [comment]
Going to also add a `slug` field to the base content.  I'm trying to figure out how I want to handle doing a `get_by_full_slug` that pays attention to the `primary_section.full_slug` in addition to the base slug on a content item though.  My first thought was to use `contribute_to_class` to add it to the the manager, but that's kind of a pain.  There's no nice clean way to add methods to the default manager.  Might end up with an extra custom manager that's added to `ContentBase` that will do the querying, but then it becomes a matter of what to name it.

**Pull request merged**